lib/pages/chat/seen_by_row.dart

```diff
@@ -26,85 +29,63 @@ class SeenByRow extends StatelessWidget {
-                size: MessageTimeStyle.seenByRowIconSize,
-              )
-            : Icon(
-                Icons.done_all,
-                color: timelineOverlayMessage
-                    ? Colors.white
-                    : Theme.of(context).colorScheme.primary,
-                size: MessageTimeStyle.seenByRowIconSize,
-              );
-    // return Container(
-    //   width: double.infinity,
-    //   alignment: Alignment.center,
-    //   child: AnimatedContainer(
-    //     constraints:
-    //         const BoxConstraints(maxWidth: FluffyThemes.columnWidth * 2.5),
-    //     height: seenByUsers.isEmpty ? 0 : 24,
-    //     duration: seenByUsers.isEmpty
-    //         ? Duration.zero
-    //         : FluffyThemes.animationDuration,
-    //     curve: FluffyThemes.animationCurve,
-    //     alignment: controller.timeline!.events.isNotEmpty &&
-    //             controller.timeline!.events.first.senderId ==
-    //                 Matrix.of(context).client.userID
-    //         ? Alignment.topRight
-    //         : Alignment.topLeft,
-    //     padding: const EdgeInsets.only(left: 8, right: 8, bottom: 4),
-    //     child: Wrap(
-    //       spacing: 4,
-    //       children: [
-    //         ...(seenByUsers.length > maxAvatars
-    //                 ? seenByUsers.sublist(0, maxAvatars)
-    //                 : seenByUsers)
-    //             .map(
-    //               (user) => Avatar(
-    //                 mxContent: user.avatarUrl,
-    //                 name: user.calcDisplayname(),
-    //                 size: 16,
-    //                 fontSize: 9,
-    //               ),
-    //             )
-    //             .toList(),
-    //         if (seenByUsers.length > maxAvatars)
-    //           SizedBox(
-    //             width: 16,
-    //             height: 16,
-    //             child: Material(
-    //               color: Theme.of(context).colorScheme.background,
-    //               borderRadius: BorderRadius.circular(32),
-    //               child: Center(
-    //                 child: Text(
-    //                   '+${seenByUsers.length - maxAvatars}',
-    //                   style: const TextStyle(fontSize: 9),
-    //                 ),
-    //               ),
-    //             ),
-    //           ),
-    //       ],
-    //     ),
-    //   ),
-    // );
+
+    return getEventIcon(context, eventStatus, seenByUsers);
+  }
+
+  MessageStatus getMessageStatus(
+    BuildContext context,
+    EventStatus? eventStatus,
+    List<User> seenByUsers,
+  ) {
+    if (eventStatus == null || eventStatus == EventStatus.sending) {
+      return MessageStatus.sending;
+    }
+
+    if (eventStatus == EventStatus.sent || seenByUsers.isEmpty) {
+      return MessageStatus.sent;
+    }
+
+    if (seenByUsers.length == controller.room!.getParticipants().length) {
+      return MessageStatus.hasBeenSeenByAll;
+    }
+
+    return MessageStatus.hasBeenSeenByOne;
```

Comment:
IMO: You should change 
```
if (conditionA) {}
if (conditionB) {}
if (conditionC) {}
```

Reply:
I'm not checking the same value everytime, I don't think it's possible to cross variables in cases